engine/src/main/java/org/destinationsol/systems/DefaultDestructionSystem.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 The Terasology Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.systems;
+
+import org.destinationsol.entitysystem.EventReceiver;
+import org.destinationsol.events.DestroyEvent;
+import org.destinationsol.events.RemovalForOptimizationEvent;
+import org.terasology.gestalt.entitysystem.entity.EntityRef;
+import org.terasology.gestalt.entitysystem.event.EventResult;
+import org.terasology.gestalt.entitysystem.event.ReceiveEvent;
+
+/**
+ * Removes an entity that receives a {@link DestroyEvent} or a {@link RemovalForOptimizationEvent}.
+ */
+public class DefaultDestructionSystem implements EventReceiver {
+
+    /**
+     * Deletes a destroyed entity.
+     */
+    @ReceiveEvent
+    public EventResult onDestroy(DestroyEvent event, EntityRef entity) {
+        entity.delete();
+        return EventResult.COMPLETE;
+    }
+
+    /**
+     * Deletes an entity that no longer needs to exist.
+     */
+    @ReceiveEvent
+    public EventResult onRemovalForOptimization(RemovalForOptimizationEvent event, EntityRef entity) {
+        System.out.println("called!");
```

Comment:
Should this still be here?

Reply:
Nope! That was for debugging purposes XD